### ISSUE TITLE
Renesas-RA: Pin-IRQ does not behave as expected.

### ISSUE THREAD

**Issue by robert-hh** [micropython/micropython#15272]
### Port, board and/or hardware

Rensas-RA, Board RA6M2

### MicroPython version

MicroPython v1.24.0-preview.36.gdc9e733ad on 2024-06-12; EK-RA6M2 with RA6M2

### Reproduction

```
from machine import Pin, idle

p5=Pin("P005", Pin.IN)
p6=Pin("P006", Pin.OUT, value=0)

def cb(pin):
    global p6
    p6(1)
    p6(0)
    
p5.irq(handler=cb, trigger=Pin.IRQ_FALLING)

while True:
    idle()
```

### Expected behaviour

Response pulse at Pin P006.

### Observed behaviour

No pulse. If I call the callback directly, I get a pulse.

### Additional Information

No, I've provided everything above.

### Code of Conduct

Yes, I agree

**Comment by TakeoTakahashi2020** [comment]
@robert-hh san, Thank you for your report. I will check.

**Comment by TakeoTakahashi2020** [comment]
@robert-hh san, I have made a PR #15283 to fix this issue.
I appreciate it if you could check it.  Thanks.

**Comment by robert-hh** [comment]
Thank you. I will try it later today.
  

  
  
  
  
>

**Comment by robert-hh** [comment]
@TakeoTakahashi2020 Pin.irq() works now in my test as expected. It triggers at and does so the right time.
Thank you for the fix.

**Comment by TakeoTakahashi2020** [comment]
@robert-hh san, Thank you very much for your check.

**Comment by dpgeorge** [comment]
Fixed by 6381ad5d0c4c594dbb8feea3bebf6222fd78f3a1
